### PR TITLE
chore: add payload types for autocomplete requests

### PR DESCRIPTION
**Pull request by srikanthccv**
Part of #2236 

This adds types to be used for the requests from the client to query the aggregate attributes and filter key-values auto-complete endpoints. Each signal has its own implementation of what and how long it looks back in time to suggest the autocomplete results. The response will be based on the provided `DataSource` and `AggregateOperator`. 

https://github.com/SigNoz/signoz/blob/8ebb76bd0c40ea7e3cb4b0e3a3f216c204447f67/pkg/query-service/model/queryParams.go#L116-L123

https://github.com/SigNoz/signoz/blob/8ebb76bd0c40ea7e3cb4b0e3a3f216c204447f67/pkg/query-service/model/queryParams.go#L79-L114

Not every signal has to support all the aggregators. The parse will look at the data source and returns validation errors.

### CODE REVIEW

**Comment by srikanthccv** [comment]
I am converting this to a draft and re-open it when everyone also agrees upon the request and response payload struct.

**Comment by srikanthccv** [comment]
PTAL

File: pkg/query-service/model/v3/v3.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	SearchText string            `json:"searchText"`
+	Limit      int               `json:"limit"`
+}
+
+// FilterAttributeKeyRequest is a request to fetch possible attribute keys
+// for a selected aggregate operator and aggregate attribute and search text.
+type FilterAttributeKeyRequest struct {
+	DataSource         DataSource        `json:"dataSource"`
+	AggregateOperator  AggregateOperator `json:"aggregateOperator"`
+	AggregateAttribute string            `json:"aggregateAttribute"`
```
Comment (nityanandagohain) [inline pkg/query-service/model/v3/v3.go:93]: `TagType            TagType           `json:"tagType"`

Comment (nityanandagohain) [inline pkg/query-service/model/v3/v3.go:93]: `TagType            TagType           `json:"tagType"`

File: pkg/query-service/model/v3/v3.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+}
+
+// FilterAttributeValueRequest is a request to fetch possible attribute values
+// for a selected aggregate operator, aggregate attribute, filter attribute key
+// and search text.
+type FilterAttributeValueRequest struct {
+	DataSource         DataSource        `json:"dataSource"`
+	AggregateOperator  AggregateOperator `json:"aggregateOperator"`
+	AggregateAttribute string            `json:"aggregateAttribute"`
+	FilterAttributeKey string            `json:"filterAttributeKey"`
```
Comment (nityanandagohain) [inline pkg/query-service/model/v3/v3.go:105]: `TagType            TagType           `json:"tagType"`

**Comment by srikanthccv** [comment]
Added the suggested changes.

**Review by nityanandagohain — APPROVED** [review]
LGTM

File: pkg/query-service/model/v3/v3.go [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+	Operator   AggregateOperator `json:"aggregateOperator"`
+	SearchText string            `json:"searchText"`
+	Limit      int               `json:"limit"`
+}
+
+type TagType string
+
+const (
+	TagTypeColumn   TagType = "column"
+	TagTypeTag      TagType = "tag"
```
Comment (nityanandagohain) [inline pkg/query-service/model/v3/v3.go:92]: Are we using `tag` and `attributes` interchangeably

Comment (makeavish) [reply pkg/query-service/model/v3/v3.go:92]: Yes

**Review by makeavish — APPROVED** [review]

**Comment by srikanthccv** [comment]
And the "tag" is smaller and occupies less space in frontend.

**Pull request merged**